backend/app/main.py

```diff
@@ -28,13 +33,66 @@
     ],
 )
 
+@app.get("/listing", response_model=ListingsGetResponseItem,
+         responses={
+            '400': {'model': ListingGetResponse},
+            '404': {'model': ListingGetResponse1},
+         })
+def get_listing(listingid: str = Query(..., description="Listing ID to retrieve")):
+    """
+    Retrieve a single listing by its listing ID
+    """
+    try:
+        # Validate listingid as a valid MongoDB ObjectId
+        if not ObjectId.is_valid(listingid):
+            raise HTTPException(status_code=400, detail="Invalid listing ID format.")
```

Comment:
When an error is raised like this, does the response match the response model we have? If not, you might have to use a custom exception handler.